File: Mavim-apps/Mavim.frontend/projects/mavim-app/src/shared/field/mappers/decimalfield.mapper.ts
import { Field } from '../models/field.model';
import { SingleDecimalField } from '../models/api';
import { GenericDecimalFieldMapper } from './abstract';
import { mapToApiFieldType, mapToFieldType } from './fieldvaluetype';

export class DecimalFieldMapper extends GenericDecimalFieldMapper<SingleDecimalField> {
	public get getUrlTypeName(): string {
		return 'decimal';
	}

	public getMappedField(field: SingleDecimalField): Field {
		return this.mapField(field);
	}

	public getMappedApiField(field: Field): SingleDecimalField {
		return this.mapApiField(field);
	}

	protected mapApiField(field: Field): SingleDecimalField {
		return {
			...field,
			fieldValueType: mapToApiFieldType(field.fieldValueType),
			data: field.data ? this.mapToDecimal(field.data[0]) : undefined
		};
	}

	protected mapField(field: SingleDecimalField): Field {
		return {
			...field,
			fieldValueType: mapToFieldType(field.fieldValueType),
			options: undefined,
			data: [this.mapToString(field.data)]
		} as Field;
	}
}

File: Mavim-services/Topic/Mavim.Manager.Api.Topic.Repository/v1/Mappers/DecimalFieldMapper.cs
using Mavim.Manager.Api.Topic.Repository.Interfaces.v1.enums;
using Mavim.Manager.Api.Topic.Repository.Interfaces.v1.Fields;
using Mavim.Manager.Api.Topic.Repository.v1.Fields;
using Mavim.Manager.Api.Topic.Repository.v1.Mappers.Abstract;
using System;

namespace Mavim.Manager.Api.Topic.Repository.v1.Mappers
{
    internal class DecimalFieldMapper : GenericDecimalFieldMapper<Model.ISingleValueSimpleField, ISingleDecimalField>
    {
        /// <summary>
        /// Gets the generic mapped decimal service field.
        /// </summary>
        /// <param name="field">The field.</param>
        /// <returns></returns>
        protected override ISingleDecimalField GetGenericMappedServiceField(Model.ISingleValueSimpleField field)
        {
            if (field == null) throw new ArgumentNullException(nameof(field));

            return new SingleDecimalField
            {
                FieldId = field.FieldDefinition.ID,
                FieldName = field.FieldDefinition.Name,
                SetOrder = field.FieldSetDefinition.OrderNumber,
                Order = field.OrderNumber,
                FieldSetId = field.FieldSetDefinition.ID,
                SetName = field.FieldSetDefinition?.Name ?? string.Empty,
                FieldValue = field.FieldValue?.ToString(),
                FieldValueType = FieldType.Decimal,
                Required = field.FieldDefinition.Required,
                Readonly = field.ReadOnly
            };
        }

        /// <summary>
        /// Gets the generic mapped decimal repo field.
        /// </summary>
        /// <param name="field">The field.</param>
        /// <param name="simpleField">The simple field.</param>
        /// <returns></returns>
        protected override object[] GetGenericMappedRepoField(ISingleDecimalField field, Model.ISimpleField simpleField = null)
        {
            return new object[] { field?.FieldValue };
        }
    }
}
